src/v/kafka/server/materialized_partition.h

```diff
@@ -39,7 +39,7 @@ class materialized_partition final : public kafka::partition_proxy::impl {
     bool is_leader() const final { return _partition->is_leader(); }
 
     ss::future<result<model::offset>> linearizable_barrier() {
-        return _partition->linearizable_barrier().then(
+        return _partition->source_partition()->linearizable_barrier().then(
```

Comment:
it looks like `_partition` is `ss::lw_shared_ptr<cluster::partition>` but I don't see `source_partition` method on `cluster::partition`. what am I missing?

Reply:
It was not added in the initial commit, it was added later in the commit this line touches.

Reply:
let's try to make sure our commits compile

Reply:
It compiles because `source_partition()` is added in this very commit, check out `partition.h` in commit `89bff825c02ba90fad70d7cbc6bc8da0f86d60c0`

Reply:
I see your point here, the type wasn't modified by this commit, nice find